spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/GridFsTemplate.java

```diff
@@ -247,28 +246,15 @@ public GridFsResource getResource(GridFSFile file) {
      * (non-Javadoc)
      * @see org.springframework.core.io.support.ResourcePatternResolver#getResources(java.lang.String)
      */
-	public GridFsResource[] getResources(String locationPattern) {
-
-		if (!StringUtils.hasText(locationPattern)) {
-			return new GridFsResource[0];
-		}
+    public GridFsResource[] getResources(String locationPattern) {
+        Pattern pattern = Pattern.compile(locationPattern);
```

Comment:
The new code skips treating the location as `AntPath` and expects it to be a regex expression right away. This is clearly breaking the contract, isn't it?

Reply:
Yes, you are right, i misused the possibility to instatiate a GridFsResource with a GridFsFile. I thought this would also provide a valid GridFsResource, but as i looked into it seems that the inputStream is missing. I will have a look. Thx for your feedback!

Reply:
Yes, it is. I will come up with a new solution, working on with your feedback.